File: Python/2.py
# wap to check whether a number is Armstrong or not.

def is_armstrong_number(number):
    # Convert the number to a string to find the number of digits
    num_str = str(number)
    
    # Calculate the number of digits
    num_digits = len(num_str)
    
    # Calculate the sum of the digits, each raised to the power of the number of digits
    armstrong_sum = sum(int(digit) ** num_digits for digit in num_str)
    
    # Check if the sum is equal to the original number
    return armstrong_sum == number

# Input from the user
num = int(input("Enter a number: "))

# Check if it's an Armstrong number and display the result
if is_armstrong_number(num):
    print(f"{num} is an Armstrong number.")
else:
    print(f"{num} is not an Armstrong number.")

File: JAVA/2.java
//write a program to enter number from user and check whether it is Armstrong or palindrome.

import java.util.Scanner;

public class NumberCheck {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int number = scanner.nextInt();

        int originalNumber = number;
        int reverse = 0;
        int sum = 0;
        int digit;

        // Check if the number is Armstrong
        while (number > 0) {
            digit = number % 10;
            sum += digit * digit * digit; // For Armstrong check, cube of digits
            number /= 10;
        }

        if (sum == originalNumber) {
            System.out.println(originalNumber + " is an Armstrong number.");
        } else {
            System.out.println(originalNumber + " is not an Armstrong number.");
        }

        // Check if the number is a palindrome
        number = originalNumber; // Reset number to the original value for palindrome check
        while (number > 0) {
            digit = number % 10;
            reverse = reverse * 10 + digit;
            number /= 10;
        }

        if (reverse == originalNumber) {
            System.out.println(originalNumber + " is a palindrome.");
        } else {
            System.out.println(originalNumber + " is not a palindrome.");
        }

        scanner.close();
    }
}
